src/components/lib/FlowGrid/background-container.js

```diff
@@ -45,46 +45,44 @@ export class BackgroundContainer extends Component {
       !_.isEqual(this.props.copiedRegion, nextProps.copiedRegion) ||
       !_.isEqual(this.props.selectedRegion, nextProps.selectedRegion) ||
       !_.isEqual(this.props.autoFillRegion, nextProps.autoFillRegion) ||
+      !_.isEqual(this.props.analyzedRegion, nextState.analyzedRegion) ||
       !_.isEqual(this.props.edges, nextProps.edges) ||
       !_.isEqual(this.state.rowHeights, nextState.rowHeights)
     )
   }
 
   render() {
-    const {edges, rowCount, getRowHeight, selectedRegion, copiedRegion, autoFillRegion} = this.props
+    const {edges, rowCount, getRowHeight, selectedRegion, copiedRegion, autoFillRegion, analyzedRegion} = this.props
     const {rowHeights} = this.state
 
     const columnWidth = $('.FlowGridCell') && $('.FlowGridCell')[0] && $('.FlowGridCell')[0].offsetWidth
     if (!columnWidth || !rowHeights.length) { return false }
 
     const containerHeight = rowHeights.reduce((a,b) => a + b)
 
+    const regions = [
+      [selectedRegion, 'selected'],
+      [analyzedRegion, 'analyzed'],
+      [copiedRegion, 'copied'],
+      [autoFillRegion, 'fill']
+    ]
+
     return (
       <div>
-        {selectedRegion.length === 2 &&
-          <Region
-            rowHeights={rowHeights}
-            columnWidth={columnWidth}
-            selectedRegion={selectedRegion}
-            type='selected'
-          />
-        }
-        {copiedRegion.length === 2 &&
-          <Region
-            rowHeights={rowHeights}
-            columnWidth={columnWidth}
-            selectedRegion={copiedRegion}
-            type='copied'
-          />
-        }
-        {autoFillRegion.length === 2 &&
-          <Region
-            rowHeights={rowHeights}
-            columnWidth={columnWidth}
-            selectedRegion={autoFillRegion}
-            type='fill'
-          />
-        }
+        {regions.map(region => {
+          if (region[0].length === 2) {
+            return (
+              <Region
```

Comment:
A few small things:
0. I'd rename these from regions to backgroundRegions or something to differentiate from the grid regions they contain.
1. There is a `isRegion` helper in the lcoation utils to ensure that a region is a region that you should use instead of checking the length.
2. I'd do make a `renderRegion([region, regionType])` function within this class, then structure this like: 
```
regions.map(region => isRegion(region[0]) ? this.renderRegion(region) : false)
```